cpp/ql/test/query-tests/Likely Bugs/Format/WrongTypeFormatArguments/Linux_signed_chars/a.c

```diff
@@ -0,0 +1,16 @@
+
+__attribute__((format(printf, 1, 3)))
+void myMultiplyDefinedPrintf(const char *format, const char *extraArg, ...)
+{
+	// ...
+}
+__attribute__((format(printf, 1, 3)))
+void myMultiplyDefinedPrintf2(const char *format, const char *extraArg, ...);
+
+void test_custom_printf1()
+{
+  myMultiplyDefinedPrintf("%i", "%s", 1); // GOOD
+  myMultiplyDefinedPrintf("%i", "%s", 1.0f); // BAD [NOT DETECTED]
+  myMultiplyDefinedPrintf2("%i", "%s", 1); // GOOD (we can't tell which definition is correct so we have to assume this is OK)
+  myMultiplyDefinedPrintf2("%i", "%s", 1.0f); // GOOD (we can't tell which definition is correct so we have to assume this is OK)
```

Comment:
Why is it `%s` in this file but `%f` in `b.c`? Why use `%i` instead of the usual `%d`?

Reply:
> Why is it %s in this file but %f in b.c?

Well spotted - it looks like it should be `%f` (as the expected comment for the fourth case isn't correct if we have `%s`).  Further, it appears the original version of the query actually handled this corrected test case correctly (thanks to some special logic in the ql file itself that I hadn't spotted before).  However the `NonConstantFormat.ql` query still needs the changes in this PR.  I should probably update the change note to reflect this.

> Why use %i instead of the usual %d?

Habit.  I've always thought `%i` is more intuitive.  There are other test cases involving `%d`, so I don't think coverage is an issue here.

Reply:
I've now updated the change note as well.

Note that this PR was never based on a real world bug, it was an extension of https://github.com/Semmle/ql/pull/1987 (that was based on a real world bug).

Sorry for the confusion.